modules/saas.py

```diff
@@ -0,0 +1,2542 @@
+            "benign_true_positive",
+            "malicious_true_positive",
+            "false_positive",
+        ]:
+            raise ValueError("Invalid category provided")
+
+        if title == "":
+            raise ValueError("Title cannot be empty.")
+        payload = {"title": title, "category": category}
+        return self._request(
+            method="post", url=f"{self.url}/assignment_outcomes", json=payload
+        )
+
+    def update_assignment_outcome(self, outcome_id=None, title="", category=""):
+        """
+        Update an existing custom Assignment Outcome
+        :param outcome_id:
+        :param tile: title of the new Assignment Outcome to create.
+        :param category: one of benign_true_positive, malicious_true_positive or false_positive
+        """
+        if category not in [
+            "benign_true_positive",
+            "malicious_true_positive",
+            "false_positive",
+        ]:
+            raise ValueError("Invalid category provided")
+
+        if title == "":
+            raise ValueError("Title cannot be empty.")
+
+        payload = {"title": title, "category": category}
+        return self._request(
+            method="put",
+            url=f"{self.url}/assignment_outcomes/{outcome_id}",
+            json=payload,
+        )
+
+    def delete_assignment_outcome(self, outcome_id=None):
+        """
+        Delete an existing custom Assignment Outcome
+        :param outcome_id: ID of the Assignment Outcome to delete
+        """
+        if not outcome_id:
+            raise ValueError("Must provide outcome_id.")
+        return self._request(
+            method="delete", url=f"{self.url}/assignment_outcomes/{outcome_id}"
+        )
+
+    def get_account_scoring(self, **kwargs):
+        """
+        Get account scoring
+        :param from:
+        :param limit:
+        """
+        return self._request(
+            method="get",
+            url=f"{self.url}/events/account_scoring",
+            params=self._generate_account_event_params(kwargs),
+        )
+
+    def get_account_detection(self, **kwargs):
+        """
+        Get account detection
+        :param from:
+        :param limit:
+        """
+        return self._request(
+            method="get",
+            url=f"{self.url}/events/account_detection",
+            params=self._generate_account_event_params(kwargs),
+        )
+
+    def get_audits(self, **kwargs):
+        """
+        Requires certain privs - Handle the error
+        Get audit events
+        :param event_timestamp_gte:
+        :param event_timestamp_lte:
+        :param from:
+        :param user_id:
+        :param event_object:
+        :param event_action:
+        :param limit:
+        """
+        return self._request(
+            method="get",
+            url=f"{self.url}/events/audits",
+            params=self._generate_audit_log_params(kwargs),
+        )
+
+
+class VectraSaaSClientV3_1(VectraSaaSClient):
```

Comment:
If we have the VERSION as a class level variable, you can just override VERSION3 and don't need to override the __init__() for the class.